specs/_features/maxeb_increase/capella.py

```diff
@@ -1776,6 +1790,49 @@ def process_execution_layer_withdraw_request(
     ))
 
 
+def is_consolidated(validator: Validator) -> bool:
+   validator.consolidated_to != UNSET_CONSOLIDATED_TO
+
+
+def process_consolidation(state: BeaconState, consolidation: Consolidation) -> None:
+    target_validator = state[consolidation.target_index]
+    source_validator = state[consolidation.source_index]
+
+    assert is_active_validator(target_validator) and not target_validator.slashed and not is_consolidated(target_validator)
+    assert not source_validator.slashed and not is_consolidated(source_validator)
+
+    # verify source withdrawal credentials, which have authority over validating keys
+    assert source_validator.withdrawal_credentials[:1] == ETH1_ADDRESS_WITHDRAWAL_PREFIX
+    assert source_validator.withdrawal_credentials[12:] == consolidation.source_address
+
+    # target validator accepts consolidation operation onto itself
+    # Fork-agnostic domain since address changes are valid across forks
+    domain = compute_domain(DOMAIN_CONSOLIDATION, genesis_validators_root=state.genesis_validators_root)
+    signing_root = compute_signing_root(source_validator.pubkey, domain)
+    assert bls.Verify(target_validator.pubkey, signing_root, consolidation.target_signature)
+
+    consolidated_balance = state.balances[consolidation.source_index]
+    state.balances[consolidation.target_index] += consolidated_balance
+    state.balances[consolidation.source_index] = 0
+
+    source_validator.consolidated_to = consolidation.target_index
+    source_validator.consolidated_balance = consolidated_balance
+    # Balance is not exiting the active set, do not apply churn
+    source_validator.exit_epoch = get_current_epoch(state)
+    source_validator.withdrawable_epoch = Epoch(source_validator.exit_epoch + config.MIN_VALIDATOR_WITHDRAWABILITY_DELAY)
```

Comment:
Source validator record with 0 balance should exit the validator set immediately. Otherwise it will be ejected on the next epoch transition. Since its balance is not exiting the validator set it can bypass the churn. withdrawable_epoch must be set sufficiently into the future to allow time to submit slashable offenses.